### PR TITLE
algorand watcher

### PR DESCRIPTION
**Pull request by jumpsiegel**

### CODE REVIEW

**Review by evan-gray — CHANGES_REQUESTED** [review]
This is looking pretty good. Can you fill out the `INITIAL_DEPLOYMENT_BLOCK_BY_CHAIN` in consts for Algorand - where the value should be the "fromBlock" to start indexing from as the deployment of the wormhole core contract (to ensure the watcher can backfill every message)

**Review by evan-gray — COMMENTED** [review]
Couple small things, I think the file name should match the class name so `Algorand.ts` -> `AlgorandWatcher.ts`, and we've been trying to have the commit names prefixed with the folder (like the wormhole repo). Admittedly, would be good to include these in a contributing doc 😛 I'm running it now and will update / approve soon!

**Review by evan-gray — CHANGES_REQUESTED** [review]
After running this for a while, I started getting the following error:
```
AlgorandWatcher.ts:343
    for (const stxn of b.block.txns) {
                               ^
TypeError: b.block.txns is not iterable
```

File: watcher/src/watchers/AlgorandWatcher.ts [hunk 1]
```diff
@@ -0,0 +1,374 @@
...
+    for (let i = 0; i < arrs.length; i++) {
+      c.set(arrs[i], offset);
+      offset += arrs[i].length;
+    }
+
+    return c;
+  }
+}
+
+export class AlgorandWatcher extends Watcher {
```
Comment (heyitaki) [inline watcher/src/watchers/AlgorandWatcher.ts:235]: Can we hoist exported consts for visibility?

File: watcher/src/watchers/Algorand.ts [hunk 2]
```diff
@@ -0,0 +1,374 @@
+// npx pretty-quick
```
[the remark this answers is not in the document]
Comment (evan-gray) [reply watcher/src/watchers/Algorand.ts]: fixed

File: watcher/src/watchers/Algorand.ts [hunk 3]
```diff
@@ -0,0 +1,374 @@
...
+    if (mainnet) {
+      this.appid = 842125965;
+      this.algodToken = '';
+      this.algodServer = 'https://mainnet-api.algonode.cloud';
+      this.algodPort = 443;
+      this.server = 'https://mainnet-idx.algonode.cloud';
+      this.port = 443;
+      this.token = '';
+    } else {
+      this.appid = 4;
```
[the remark this answers is not in the document]
Comment (evan-gray) [reply watcher/src/watchers/Algorand.ts]: fixed

File: watcher/src/watchers/AlgorandWatcher.ts (hunk 1 above)
Comment (evan-gray) [reply watcher/src/watchers/AlgorandWatcher.ts:232]: fixed

**Comment by evan-gray** [comment]
testing after rebase and still hitting
```
[2022-12-27 15:16:13.160 +0000] [info] [algorand] fetching info for blocks 23761194 to 23761198
[2022-12-27 15:16:13.309 +0000] [error] [algorand] b.block.txns is not iterable
```

**Comment by evan-gray** [comment]
Investigating https://algoexplorer.io/block/23761195

**Review by evan-gray — APPROVED** [review]
Seems to be running smoothly after rebase with the empty block fix

**Pull request merged**